src/transaction.rs

```diff
@@ -613,10 +615,25 @@ impl Transaction {
         bitcoin::Wtxid::from_engine(enc)
     }
 
-    /// Get the total transaction fee.
-    pub fn fee(&self) -> u64 {
+    /// Get the total transaction fee in the given asset.
+    pub fn fee_in(&self, asset: AssetId) -> u64 {
+        let asset = confidential::Asset::Explicit(sha256d::Hash::from_inner(asset.into_inner().into_inner()));
         // All values should be explicit, so we don't have to assert that here.
-        self.output.iter().filter(|o| o.is_fee()).filter_map(|o| o.value.explicit()).sum()
+        self.output.iter()
+            .filter(|o| o.is_fee() && o.asset == asset)
+            .filter_map(|o| o.value.explicit())
+            .sum()
+    }
+
+    /// Get all fees in all assets.
+    pub fn all_fees(&self) -> HashMap<AssetId, u64> {
+        let mut fees = HashMap::new();
+        for out in self.output.iter().filter(|o| o.is_fee()) {
+            let asset = AssetId::from_inner(sha256::Midstate::from_inner(out.asset.explicit().unwrap().into_inner()));
```

Comment:
maybe a comment why is safe to unwrap() ?

Reply:
good point, I have that comment in fee_in, but not here, will add.